kvm: tun: give the network interface a name

alban opened the pull request:
I tried to give the tap device a name so it's obvious it was created by rkt and that there would be a way for NetworkManager to ignore it. But the current patch does not work.

See also https://github.com/coreos/rkt/issues/1302

/cc @ppalucki @jellonek

jellonek commented:
What you mean by "current patch does not work"?
For me it works as expected:
> ip l | grep tap
> 70: rkttap0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 500

alban commented:
@jellonek : thanks for the review. It works for me now. I guess I tested it wrongly.

I renamed the variable to `nameTemplate`, added 4 characters from the pod UUID and improved the commit message:

>    kvm: tun: give the network interface a name
>    
>    Example of name: rkt-7543-tap0
>    
>    The name contains "rkt" so it could be used by NetworkManager or ConnMan
>    to know it should ignore it.
>    See discussion in https://github.com/coreos/rkt/issues/1302
>    
>    It contains 4 characters from the pod UUID. That should make it easier
>    to debug which network device is used by a pod. It does not contain the
>    full UUID because the device name is limited to 16 characters.

jonboulle commented:
LGTM

alban commented:
Thanks!
